packages/dart/noports_core/lib/src/srv/srv_impl.dart

```diff
@@ -792,29 +843,49 @@ class SrvImplDart implements Srv<SocketConnector> {
       List<String> requests = eventStr.split('connect:');
       for (String request in requests) {
         if (request.isNotEmpty) {
-          await _handleMultiConnectRequest(sc, hosts, 'connect:$request');
+          await _handleControlRequest(sc, relayAddress, 'connect:$request');
         }
       }
+    } catch (e, st) {
+      logger.shout('Caught (will rethrow) error: $e\nStack Trace:\n$st');
+      rethrow;
     } finally {
       controlStreamMutex.release();
     }
   }
 
+  Future<InternetAddress> resolveRequestedLocalHost() async {
```

Comment:
- `InternetAddress.lookup`'s `type` parameter defaults to `InternetAddressType.any`
- Per that method's documentation, the order of the returned list is not guaranteed. 
- We will try to resolve via IPv4 first; if that doesn't work then we will try to resolve via IPv6
- We have more work to do in this area, since ipv6 addresses contain colons and we have code scattered around which parses things based on colons. In general we are not properly supporting IPv6. I'll create a ticket for remedying that